src/sentry/api/endpoints/relay_register.py

```diff
@@ -95,11 +95,9 @@ def post(self, request):
                 status=status.HTTP_400_BAD_REQUEST,
             )
 
-        try:
-            relay = Relay.objects.get(relay_id=relay_id)
-        except Relay.DoesNotExist:
-            pass
-        else:
+        relay, static = relay_from_id(request, relay_id)
```

Comment:
I clicked a bit too early, there's one more potential issue with this line. A few lines above there's a check for `SENTRY_RELAY_OPEN_REGISTRATION` that is meant to also cover the statically configured Relays. We should ensure that we do not bail out if the flag is set to false but someone is registering one of the new statically configured relays.

Reply:
Yes... I think that was the reason we had the check for static relays in the original is_internal_relay implementation...